src/jenkins_job_insight/storage.py

```
@@ -2032,6 +2032,73 @@ async def override_classification(
+                " ORDER BY id DESC LIMIT 1",
+                [test_name, job_id, _child_job_name, _child_build_number],
+            )
+        ).fetchone()
+        if override_row and override_row[0]:
+            return override_row[0]
+
+        # 2. Fall back to failure_history
+        fh_query = (
+            "SELECT classification FROM failure_history"
+            " WHERE job_id = ? AND test_name = ?"
+            f" AND classification IN {_history_classifications_sql}"
```

Ruff (0.15.9)

[error] Possible SQL injection vector through string-based query construction

(S608)